### PR TITLE
New Barracks Setup

### LINKED ISSUE #7212: Mirroring of barracks tower placeholder doesn't work
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: 0.14.210-alpha
- Structurize Version: 0.13.177-alpha
- Related Mods and their Versions:

## Expected behavior

When you place a structure with a barracks tower placeholder (eg. medieval birch alternative barracks) mirrored, the barracks tower hut blocks will be in the correct mirrored state, and the barracks tower will be built mirrored.

## Actual behavior

Rotation works, but preview shows the barracks tower placeholder in the wrong orientation, and actually building the barracks tower puts it in unmirrored.

## Steps to reproduce the issue

1. Grab a barracks hut block and begin placing it in a style that has barracks towers inside
2. Find a barracks tower in the preview and observe its orientation.  Mirror the barracks.
3. Find the corresponding barracks tower hut block and observe that it is in a different orientation relative to the rest of the structure
4. Build the barracks mirrored
5. Attempt to build the barracks tower.  Notice the construction tape is in the wrong location.

### Logs

- latest.log: Unlike the other issues I'm uploading now, this does show up a bit in latest.log, but I think not usefully.  https://gist.github.com/dhouck/ae2c7f039644daad6452fb300da7b28d, relevant bit starting around 1:54:36

### Notes

I'm not fully sure what causes this.  The `mirror` function of BlockBarracksTowerSubstitution.java looks reasonable so I'd at least expect this to show up correctly in the previews.  I notice that `AbstractBlockHut.java` doesn't have a `mirror` function even though it does have `rotate`, and a bit more work might need to be done to pass the mirror bit down to the hut block's building, though.

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #7212 
Closes #
Closes #

# Changes proposed in this pull request:
- Remove barracks placeholder and solve barracks issue
-
-

Review please

### CODE REVIEW

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 1]
```diff
@@ -1963,8 +1961,6 @@
   "block.minecolonies.blockpostbox": "Postbox",
   "block.minecolonies.blockstash": "Stash",
   "block.minecolonies.blockstash.name": "Stash",
-  "block.minecolonies.blocksubstitution": "Placeholder Block",
-  "block.minecolonies.blocksubstitution.name": "Placeholder Block",
```
Comment (uecasm) [inline src/main/resources/assets/minecolonies/lang/en_us.json:1967]: This doesn't seem intended.

**Review by uecasm — CHANGES_REQUESTED** [review]
Presumably new barracks scans are needed before this can be merged?  Because I haven't seen any updates to those yet, and they still currently require the placeholders...

File: src/main/resources/assets/minecolonies/lang/en_us.json (hunk 1 above)
Comment (Raycoms) [reply src/main/resources/assets/minecolonies/lang/en_us.json:1967]: Those don't exist in minecolonies anymore, were moved to structurize a long time ago.

**Comment by Raycoms** [comment]
> Presumably new barracks scans are needed before this can be merged? Because I haven't seen any updates to those yet, and they still currently require the placeholders...

Yes, I haven't included them in the PR yet, because else it will be impossible to review

**Comment by uecasm** [comment]
Wouldn't be impossible to review, you can hide files by type or only look at specific commits.  But it's not possible to test or merge without updated schematics.

**Comment by Raycoms** [comment]
> Wouldn't be impossible to review, you can hide files by type or only look at specific commits. But it's not possible to test or merge without updated schematics.

It's because if it's too many files github "crashes" and you can't filter anymore. I'm also still missing a few files, when the code was approved I'll merge the files in here so people can do testing.

**Review by uecasm — DISMISSED** [review]
Well, there's not a lot of code otherwise to review 😀

My only concern is that without the placeholder intermediate step there might not be anything calling `setStyle` correctly, but hopefully the integrated building stuff takes care of that properly.  Existing evidence suggests that we shouldn't be relying on the schematics to have that correct already...

**Comment by uecasm** [comment]
Have you let the maker of the bunker style know about this change?

**Comment by Raycoms** [comment]
Just did, good you remembered =D

**Pull request merged**